public/js/actions/sources.js

```diff
@@ -184,6 +184,25 @@ function togglePrettyPrint(id) {
   };
 }
 
+function loadOriginalSourceText(source, getState, dispatch) {
+  return Task.spawn(function* () {
+    const generatedSource = getSource(
+      getState(),
+      getGeneratedSourceId(source)
+    );
+
+    const generatedSourceText = yield dispatch(
+      loadSourceText(generatedSource.toJS())
```

Comment:
I believe we need the generated source to get the original source. I'm getting the original source from the source node. I might be mistaken and can read each of the original sources from the consumer...